### ISSUE TITLE
The current gradle version is extremely old

### ISSUE THREAD

**Issue by ParkerM** [spring-projects/spring-ldap#581]
I recently cloned this repo to explore an issue, ran `./gradlew clean build`, and was caught off guard by the sight of:
```
Downloading https://services.gradle.org/distributions/gradle-3.1-all.zip
```

Gradle 3.1 was published [nearly 5 years ago](https://github.com/gradle/gradle/releases/tag/v3.1.0), and other gradle-based projects in the [spring-projects](https://github.com/spring-projects) org seem to diligently abide gradle's release cadence. 

I'd be happy to submit a PR to migrate towards a maintained version of gradle, but there are a bunch of bespoke buildscripts in the [gradle/](https://github.com/spring-projects/spring-ldap/tree/e27bd78395a304aff1ccd2936096cb8eecf14b8a/gradle) and [buildSrc/](https://github.com/spring-projects/spring-ldap/tree/e27bd78395a304aff1ccd2936096cb8eecf14b8a/buildSrc) directories that would probably break as a result.

**Comment by jzheaux** [comment]
I've been able to update to Gradle 6.9 in commits 133cd3e846ac23c97e71d20af12131e733d9d97a, 9655d05127240f31dc819191b8644ef7dab95111, and d0336148a3b57e24089797ec928e76866978598d.

To get to 7, we'll need to replace `MergePlugin` with something else that can generate a merged jar between `test-support` and `test-support-unboundid`. I haven't looked very far into it yet, but https://github.com/spring-projects/spring-security/blob/main/saml2/saml2-service-provider/spring-security-saml2-service-provider.gradle may provide some guidance.

Are you interested in helping update to Gradle 7?

**Comment by jzheaux** [comment]
Closed via [28f2ca6](https://github.com/spring-projects/spring-ldap/commit/28f2ca6ee0a3e87df3afede1ec9aae1a36201197).
